tests/components/plex/test_config_flow.py

```diff
@@ -502,3 +530,161 @@ async def test_multiple_servers_with_import(hass):
+
+    # Advanced automatic
+    result = await hass.config_entries.flow.async_init(
+        config_flow.DOMAIN, context={"source": "user", "show_advanced_options": True}
+    )
+
+    assert result["data_schema"] is not None
+    assert result["type"] == "form"
+    assert result["step_id"] == "user_advanced"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input={"setup_method": AUTOMATIC_SETUP_STRING}
+    )
+
+    assert result["type"] == "external"
+    hass.config_entries.flow.async_abort(result["flow_id"])
+
+    # Advanced manual
+    result = await hass.config_entries.flow.async_init(
+        config_flow.DOMAIN, context={"source": "user", "show_advanced_options": True}
+    )
+
+    assert result["data_schema"] is not None
+    assert result["type"] == "form"
+    assert result["step_id"] == "user_advanced"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input={"setup_method": MANUAL_SETUP_STRING}
+    )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "manual_setup"
+
+    mock_plex_server = MockPlexServer()
+
+    MANUAL_SERVER = {
+        CONF_HOST: MOCK_SERVERS[0][CONF_HOST],
+        CONF_PORT: MOCK_SERVERS[0][CONF_PORT],
+        CONF_SSL: False,
+        CONF_VERIFY_SSL: True,
+        CONF_TOKEN: MOCK_TOKEN,
+    }
+
+    MANUAL_SERVER_NO_HOST_OR_TOKEN = {
+        CONF_PORT: MOCK_SERVERS[0][CONF_PORT],
+        CONF_SSL: False,
+        CONF_VERIFY_SSL: True,
+    }
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input=MANUAL_SERVER_NO_HOST_OR_TOKEN
+    )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "manual_setup"
+    assert result["errors"]["base"] == "host_or_token"
+
+    with patch(
+        "plexapi.server.PlexServer", side_effect=requests.exceptions.SSLError,
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input=MANUAL_SERVER
+        )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "manual_setup"
+    assert result["errors"]["base"] == "ssl_error"
+
+    with patch(
+        "plexapi.server.PlexServer", side_effect=WrongCertValidaitionException,
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input=MANUAL_SERVER
+        )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "manual_setup"
+    assert result["errors"]["base"] == "ssl_error"
+
+    with patch(
+        "homeassistant.components.plex.PlexServer.connect",
+        side_effect=requests.exceptions.SSLError,
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input=MANUAL_SERVER
+        )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "manual_setup"
+    assert result["errors"]["base"] == "ssl_error"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5462445)